db/commitlog/commitlog.cc

```diff
@@ -264,6 +264,44 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
     using request_controller_units = semaphore_units<timeout_exception_factory, db::timeout_clock>;
     request_controller_type _request_controller;
 
+    class named_file : public seastar::file {
+        sstring _name;
+        uint64_t _known_size = 0;
+    public:
+        named_file(std::string_view name)
+            : _name(name)
+        {}
+        named_file(named_file&&) = default;
+
+        future<> open(open_flags, file_open_options) noexcept;
+        future<> rename(std::string_view to);
+
+        // NOT overrides. Nothing virtual. Will rely on exact type
+        template <typename CharType>
+        future<size_t> dma_write(uint64_t pos, const CharType* buffer, size_t len, const io_priority_class& pc = default_priority_class(), io_intent* intent = nullptr) noexcept;
+        future<size_t> dma_write(uint64_t pos, std::vector<iovec> iov, const io_priority_class& pc = default_priority_class(), io_intent* intent = nullptr) noexcept;
+
+        future<> truncate(uint64_t length) noexcept;
+        future<> allocate(uint64_t position, uint64_t length) noexcept;
+        future<> remove_file() noexcept;
+
+        void assign(file&& f) {
+            this->file::operator=(std::move(f));
```

Comment:
Shouldn't this do something about _known_size?

Reply:
When we do this we are assuming size will be updated elsewhere or is already correct (only used in allocate_ex file wrapping and named_file::open(...). The type is not 100% self-contained in this. Should maybe make the `assign` take a size parameter as well, so caller must provide it.

Reply:
Ok, since it's just an internal utility class, I guess it doesn't need to be 100% beautiful. So ok.